cmd/end2endtest/helpers.go

```diff
@@ -631,3 +631,53 @@ func cspGenProof(pid, voterKey []byte, csp *ethereum.SignKeys) (*apiclient.Censu
 
 	return &apiclient.CensusProof{Proof: caProofBytes}, nil
 }
+
+func (t *e2eElection) verifyVoteCount(startTime time.Time, nvotesExpected int) error {
+	// Wait for all the votes to be verified
+	log.Infof("waiting for all the votes to be registered...")
+	for {
+		count, err := t.api.ElectionVoteCount(t.election.ElectionID)
+		if err != nil {
+			log.Warn(err)
+		}
+		if count == uint32(nvotesExpected) {
+			break
+		}
+		time.Sleep(time.Second * 5)
+		log.Infof("verified %d/%d votes", count, nvotesExpected)
+		if time.Since(startTime) > t.config.timeout {
+			return fmt.Errorf("timeout waiting for votes to be registered")
+		}
+	}
+
+	log.Infof("%d votes registered successfully, took %s (%d votes/second)",
+		t.config.nvotes, time.Since(startTime), int(float64(nvotesExpected)/time.Since(startTime).Seconds()))
+
+	return nil
+}
+
+func (t *e2eElection) setElectionEnd() error {
```

Comment:
is there any case where you want to end an election and not waitUntilElectionResults?

if not, i'd merge both funcs into a `endElectionAndFetchResults`